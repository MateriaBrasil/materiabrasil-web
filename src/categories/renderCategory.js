import React from 'react'
import isEmpty from 'lodash/isEmpty'
import negate from 'lodash/negate'
import find from 'lodash/find'
import map from 'lodash/map'
import includes from 'lodash/includes'
import Typography from '@material-ui/core/Typography'
import Paper from '@material-ui/core/Paper'

import Create from './Create'
import Checkbox from './Checkbox'
import Destroy from './Destroy'
import CategoryChildren from './CategoryChildren'

export const isPresent = negate(isEmpty)

const renderCategory = props => category => {
  const { materialCategories, parent, filters = {} } = props

  const materialCategory = find(
    materialCategories,
    ({ categoryId }) => categoryId === category.id,
  )

  const childProps = {
    ...props,
    key: category.id,
    parent,
    category,
    materialCategory,
  }

  const Component = materialCategory ? Destroy : Create

  const { selectCategory, unselectCategory } = filters.actions || {}

  const checked = includes(
    map(filters.selectedCategories, ({ id }) => id),
    category.id,
  )

  const filterAction = checked ? unselectCategory : selectCategory
  const action = () => filterAction(category)

  return isPresent(category.children) ? (
    category.parentId ? (
      <CategoryChildren key={category.id} {...props} category={category} />
    ) : (
      <div key={category.id} style={{ marginBottom: 20 }}>
        <Typography variant="headline" style={{ marginBottom: 20 }}>
          {category.name}
        </Typography>
        <Paper>
          {category.children.map(
            renderCategory({ ...props, parent: category }),
          )}
        </Paper>
      </div>
    )
  ) : filters.selectedCategories ? (
    <Checkbox {...childProps} action={action} checked={checked} />
  ) : (
    <Component {...childProps} />
  )
}

export default renderCategory
