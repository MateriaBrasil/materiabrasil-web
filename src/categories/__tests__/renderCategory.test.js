import renderer from 'react-test-renderer'

import renderCategory from '../renderCategory'

it('render correctly', () => {
  const props = {
    onChangeExpanded: jest.fn(),
    materialCategories: [{ id: 1234 }, { id: 21 }],
    rootCategory: { id: 321 },
  }
  const onSelect = jest.fn()
  const category = { id: '1234', name: 'foo-name', children: [] }
  const tree = renderer
    .create(renderCategory(props, onSelect)(category))
    .toJSON()
  expect(tree).toMatchSnapshot()
})
