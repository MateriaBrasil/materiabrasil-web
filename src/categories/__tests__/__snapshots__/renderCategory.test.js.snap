// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<label
  className="MuiFormControlLabel-root-1"
  style={
    Object {
      "margin": 0,
    }
  }
>
  <span
    className="MuiButtonBase-root-19 MuiIconButton-root-13 MuiSwitchBase-root-9 MuiRadio-root-4 MuiRadio-colorSecondary-8"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role={undefined}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label-18"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-22"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <input
        checked={false}
        className="MuiSwitchBase-input-12"
        disabled={undefined}
        id={undefined}
        name="category-root-321"
        onChange={[Function]}
        tabIndex={undefined}
        type="radio"
        value="foo-name"
      />
    </span>
    <span
      className="MuiTouchRipple-root-28"
    />
  </span>
  <span
    className="MuiTypography-root-35 MuiTypography-body1-44 MuiFormControlLabel-label-3"
  >
    foo-name
  </span>
</label>
`;
